MRML/vtkIGTLToMRMLPolyData.cxx

```diff
@@ -380,10 +380,9 @@ int vtkIGTLToMRMLPolyData::MRMLToIGTL(unsigned long event, vtkMRMLNode* mrmlNode
     
     //------------------------------------------------------------
     // Allocate Status Message Class
-    if (this->OutPolyDataMessage.IsNull())
-      {
-      this->OutPolyDataMessage = igtl::PolyDataMessage::New();
-      }
+    // Overwrite old message because we want to clear arrays from previous
+    // calls that will not be set anew.
+    this->OutPolyDataMessage = igtl::PolyDataMessage::New();
```

Comment:
Currently igtl::PolyDataMessage:Clear() has about the same effect as creating a new object, as it deallocates all memory. There are two options to improve performance:

A. Change behavior of igtl::PolyDataMessage:Clear() to keep allocated memory (by not deleting the m_Points, m_Vertices, etc. just calling clear() on them)

B. Add a igtl::PolyDataMessage:Reset() function (that calls clear on m_Points, m_Vertices, etc.) and call igtl::PolyDataMessage:Reset() instead of igtl::PolyDataMessage:Clear() in vtkIGTLToMRMLPolyData::MRMLToIGTL. It is similar to the Reset() function in VTK containters, such as vtkPoints, which changes the size of the container, but keeps the capacity.

Either case, it could be useful to add a Squeeze() method, which would free up unused memory (can be implemented by a std::vector:swap operation).